test/gtest/benchmark/make_xfer_req.cpp

```diff
@@ -0,0 +1,473 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file make_xfer_req.cpp
+ * @brief Dedicated benchmark for nixlAgent::makeXferReq isolated from backend implementations
+ *
+ * This benchmark uses a single agent with mock backend and loopback transfers to measure
+ * the pure overhead of makeXferReq without any real backend implementation costs.
+ *
+ * It measures:
+ * - Parameter validation
+ * - Descriptor list processing
+ * - Descriptor merging optimization
+ * - Memory allocation and copying
+ *
+ * By using loopback transfers (agent to itself), we eliminate multi-agent coordination
+ * overhead and focus purely on the makeXferReq function performance.
+ */
+
+#include "common.h"
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "mocks/gmock_engine.h"
+
+#include "nixl.h"
+#include "nixl_types.h"
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include <iomanip>
+#include <cmath>
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::NiceMock;
+using ::testing::Invoke;
+
+namespace gtest {
+
+/**
+ * @class BenchmarkMakeXferReq
+ * @brief Benchmark suite for nixlAgent::makeXferReq with mock backend isolation
+ *
+ * This test fixture creates a minimal environment with a single agent and mock backend
+ * to measure the performance of makeXferReq in isolation from any real backend implementation.
+ * Uses loopback transfers (agent to itself) to avoid multi-agent overhead.
+ */
+class BenchmarkMakeXferReq : public ::testing::Test {
+protected:
+    void
+    SetUp() override {
+        env_.addVar("NIXL_TELEMETRY_ENABLE", "n");
```

Comment:
why is it needed? We may have other vars affecting perf, e.g. NIXL_LOG_LEVEL
(also it is disabled by default)